server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -495,2084 +386,218 @@ private static long jitter() {
-        boolean isolated, long readaheadThreshold, TSamplerConfiguration tSamplerConfig,
-        long batchTimeOut, String contextArg, Map<String,String> executionHints)
-        throws NotServingTabletException, ThriftSecurityException,
-        org.apache.accumulo.core.tabletserver.thrift.TooManyFilesException,
-        TSampleNotPresentException {
-
-      TableId tableId = TableId.of(new String(textent.getTable(), UTF_8));
-      NamespaceId namespaceId;
-      try {
-        namespaceId = Tables.getNamespaceId(getContext(), tableId);
-      } catch (TableNotFoundException e1) {
-        throw new NotServingTabletException(textent);
-      }
-      if (!security.canScan(credentials, tableId, namespaceId, range, columns, ssiList, ssio,
-          authorizations)) {
-        throw new ThriftSecurityException(credentials.getPrincipal(),
-            SecurityErrorCode.PERMISSION_DENIED);
-      }
-
-      if (!security.authenticatedUserHasAuthorizations(credentials, authorizations)) {
-        throw new ThriftSecurityException(credentials.getPrincipal(),
-            SecurityErrorCode.BAD_AUTHORIZATIONS);
-      }
-
-      final KeyExtent extent = new KeyExtent(textent);
-
-      // wait for any writes that are in flight.. this done to ensure
-      // consistency across client restarts... assume a client writes
-      // to accumulo and dies while waiting for a confirmation from
-      // accumulo... the client process restarts and tries to read
-      // data from accumulo making the assumption that it will get
-      // any writes previously made... however if the server side thread
-      // processing the write from the dead client is still in progress,
-      // the restarted client may not see the write unless we wait here.
-      // this behavior is very important when the client is reading the
-      // metadata
-      if (waitForWrites) {
-        writeTracker.waitForWrites(TabletType.type(extent));
-      }
-
-      Tablet tablet = getOnlineTablet(extent);
-      if (tablet == null) {
-        throw new NotServingTabletException(textent);
-      }
-
-      HashSet<Column> columnSet = new HashSet<>();
-      for (TColumn tcolumn : columns) {
-        columnSet.add(new Column(tcolumn));
-      }
+  protected TransactionWatcher getWatcher() {
+    return watcher;
+  }
 
-      ScanParameters scanParams = new ScanParameters(batchSize, new Authorizations(authorizations),
-          columnSet, ssiList, ssio, isolated, SamplerConfigurationImpl.fromThrift(tSamplerConfig),
-          batchTimeOut, contextArg);
+  protected TabletServerLogger getLogger() {
+    return logger;
+  }
 
-      final SingleScanSession scanSession = new SingleScanSession(credentials, extent, scanParams,
-          readaheadThreshold, executionHints);
-      scanSession.scanner =
-          tablet.createScanner(new Range(range), scanParams, scanSession.interruptFlag);
+  protected String getLockID() {
+    return lockID;
+  }
```

Comment:
Makes sense. Unrelated, but I was wondering why this and some other variables weren't volatile, should they be?

Reply:
Not sure at this time. I'd have to investigate how they are used.

Reply:
> Unrelated, but I was wondering why this and some other variables weren't volatile, should they be?

If its not final and accessed by multiple threads outside of any lock then it seems like it should be.  That could be a follow on PR.

Reply:
> However, I think this one should stay, because it provides access to a non-final field

There is at least one other field that is non final and that is clientAddress.